src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
Cool! I was unfamiliar with McCabe complexity. Admittedly much of the code could use refactoring (and probably a full re-write). 

WRT the action statuses, it seems like we have good signal for whenever we know an action failed. And pretty poor signals to determine if actions succeed. Would it make sense to just indicate failure?

Also, I'm trying to understand how an agent would use the action_status signal: Presumably if my action fails then I need to try a different action in the next step? Have you found other use cases?